plugins/migration/bin/rename.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A script that automatically upgrades pieces of the Blockly API
+ *     that have been renamed by modifying developers' local files.
+ */
+'use strict';
+
+import {createSubCommand, extractRequiredInfo} from './command.js';
+import fetch from 'node-fetch';
+import {readFileSync, writeFileSync} from 'fs';
+import * as versionUtils from './versions.js';
+import JSON5 from 'json5';
+
+
+const DATABASE_URL = `https://raw.githubusercontent.com/google/blockly/develop/scripts/migration/renamings.json5`;
+
+export const rename = createSubCommand(
+    'rename', '>=5', 'Perform renamings for breaking changes')
+    .option(
+        '-i, --in-place [suffix]',
+        'do renamings in-place, optionally create backup files with the ' +
+        'given suffix. Otherwise output to stdout')
+    .action(async function() {
+      const {fromVersion, toVersion, fileNames} = extractRequiredInfo(this);
+
+      const renamer = new Renamer(await getDatabase(), fromVersion, toVersion);
+      fileNames.forEach((name) => {
+        const contents = readFileSync(name, 'utf8');
+        const newContents = renamer.rename(contents);
+        const i = this.opts().i;
+        if (i) {
+          if (typeof i == 'string') writeFileSync(name + i, contents);
+          writeFileSync(name, newContents);
+          process.stderr.write(`Migrated renamings in ${name}`);
+        } else {
+          process.stdout.write(newContents);
+        }
+      });
+    });
+
+/**
+ * Gets the database of renames.
+ * @return {!Promise<Object>} The database of renames as an object.
```

Comment:
`!Promise<!Object>`

unless parse can return null? in which case `?`

i feel like we would want some error messaging here in case the file can't be parsed, instead of just crashing